server/audit/views.py

```diff
@@ -1,69 +1,202 @@
+    records = Record.objects.filter(audit=audit_id)
+
+    missing = records.filter(status='Missing').count()
+    found = records.filter(status='Provided').count()
+    total_records_no_new = found + missing
 
-class ItemToSKViewSet(viewsets.ModelViewSet):
+    audit_accuracy = 0.0 if total_records_no_new == 0 else found / total_records_no_new
+    setattr(audit, 'accuracy', audit_accuracy)
+    audit.save()
+
+class BinToSKViewSet(viewsets.ModelViewSet):
     """
     API endpoint that allows Audits to be created.
     """
     http_method_names = ['post', 'get', 'delete']
-    queryset = ItemToSK.objects.all()
-    serializer_class = ItemToSKSerializer
+    queryset = BinToSK.objects.all()
+    permission_classes = []
 
     def get_permissions(self):
         factory = PermissionFactory(self.request)
         permission_classes = factory.get_general_permissions([
-            CheckInitAuditData, HasSameOrgInQuery, ValidateSKOfSameOrg])
+            CheckAuditOrganizationById, HasSameOrgInQuery, ValidateSKOfSameOrg])
         return [permission() for permission in permission_classes]
 
+    def get_serializer(self, *args, **kwargs):
+        serializer_class = GetBinToSKSerializer
+        if self.request.method != 'GET':
+            serializer_class = PostBinToSKSerializer
+        return serializer_class(*args, **kwargs)
+
     def create(self, request, *args, **kwargs):
         serializer = self.get_serializer(data=request.data)
         serializer.is_valid(raise_exception=True)
         saved_audit = serializer.save()
-        if saved_audit:
-            data = {'success': 'success'}
         if not saved_audit:
             return Response({'error': 'failed'}, status=status.HTTP_400_BAD_REQUEST)
-        return Response(data, status=status.HTTP_201_CREATED)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
 
     def list(self, request):
-        queryset = self.filter_queryset(self.get_queryset()).filter(
-                init_audit_id=request.query_params.get('init_audit_id'))
+        customuser_id = request.query_params.get('customuser_id')
+        init_audit_id = request.query_params.get('init_audit_id')
+
+        if customuser_id:
+            self.queryset = self.queryset.filter(customuser_id=customuser_id)
+        if init_audit_id:
+            self.queryset = self.queryset.filter(init_audit_id=init_audit_id)
+
+        serializer = self.get_serializer(self.queryset, many=True)
+        return Response(serializer.data)
+
+class RecordViewSet(viewsets.ModelViewSet):
+    http_method_names = ['post', 'get', 'patch', 'delete']
+    queryset = Record.objects.all()
+    permission_classes = []
+
+    def get_permissions(self):
+        factory = PermissionFactory(self.request)
+        permission_classes = factory.get_general_permissions(
+            [HasSameOrgInQuery, ValidateSKOfSameOrg])
+        return [permission() for permission in permission_classes]
```

Comment:
I don't see how these permission prevent people from creating, getting, or updating records across organizations.

Reply:
thats possible, i didnt change the implementation for these

Reply:
we'll have to work on this, the records are the latest addition to the PR, along with the rest of the comments the permissions seem to be the main thing to work on